### ISSUE TITLE
Format :: Severity discrepancy between gcc and json

### ISSUE THREAD

**Issue by lcrockett** [koalaman/shellcheck#2315]
#### Information
- Rule Id `N/A`
- My shellcheck version `0.7.2` `Linux Ubuntu 20.04 | @MacOS Big Sur`
- [X] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [X] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Problem description
It seems that when using `--format=gcc` `style` related feedback is marked as `note` whereas using `--format=json` `style` related feedback is correctly marked with the appropriate severity level `style`. Looking at [GCC.hs](https://github.com/koalaman/shellcheck/blob/master/src/ShellCheck/Formatter/GCC.hs#L58-L61) it seems that there is no matching for `style` related messages and everything that is not an `error` or `warning` is categorized as `note`. Not sure though if this is intentional or a bug.

In this specific scenario i'm hooking up shellcheck with `efm-langserver` that AFAIK only understands `gcc` formatting and therefore cannot identify `style` related feedback properly. I have explicitly configured `diagnostic-languageserver` for shellcheck as it does understand `json` formatting and therefore properly identifies `style` related feedback. I'm only using `diagnostic-languageserver` as a workaround for this issue but would get rid of that workaround if `gcc` style formatting would mark `style` messages properly.

#### Here's a snippet or screenshot that shows the problem:

```sh
$ shellcheck --format=gcc shell-script
shell-script:68:7: error: Double quote array expansions to avoid re-splitting elements. [SC2068]
shell-script:70:5: warning: Tilde does not expand in quotes. Use $HOME. [SC2088]
shell-script:72:6: note: Double quote to prevent globbing and word splitting. [SC2086]
shell-script:74:5: note: Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead. [SC2002]
$ shellcheck --format=json shell-script | jq
[
  {
    "file": "shell-script",
    "line": 68,
    "endLine": 68,
    "column": 7,
    "endColumn": 9,
    "level": "error",
    "code": 2068,
    "message": "Double quote array expansions to avoid re-splitting elements.",
    "fix": null
  },
  {
    "file": "shell-script",
    "line": 70,
    "endLine": 70,
    "column": 5,
    "endColumn": 18,
    "level": "warning",
    "code": 2088,
    "message": "Tilde does not expand in quotes. Use $HOME.",
    "fix": null
  },
  {
    "file": "shell-script",
    "line": 72,
    "endLine": 72,
    "column": 6,
    "endColumn": 8,
    "level": "info",
    "code": 2086,
    "message": "Double quote to prevent globbing and word splitting.",
    "fix": {
      "replacements": [
        {
          "endColumn": 6,
          "column": 6,
          "endLine": 72,
          "replacement": "\"",
          "line": 72,
          "insertionPoint": "afterEnd",
          "precedence": 7
        },
        {
          "endColumn": 8,
          "column": 8,
          "endLine": 72,
          "replacement": "\"",
          "line": 72,
          "insertionPoint": "beforeStart",
          "precedence": 7
        }
      ]
    }
  },
  {
    "file": "shell-script",
    "line": 74,
    "endLine": 74,
    "column": 5,
    "endColumn": 9,
    "level": "style",
    "code": 2002,
    "message": "Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.",
    "fix": null
  }
]
$ 
```

**Comment by koalaman** [comment]
It was intentional since these were the only three levels gcc 4 used back then, and ShellCheck's four levels had to be mapped onto them somehow. I know GCC output has changed quite a bit since then. Is there a better way to map it today?

**Comment by lcrockett** [comment]
Hmm, not sure how to map a more fitting output to current days GCC standards to be honest.

Perhaps it would make more sense introducing a new (GCC-like) formatting output (i.e. gcc1, or a completely different name) that outputs all 4 levels (error, warning, information, hint, matching up with Neovim LSP category naming) ? This way current GCC formatting stays put not breaking existing workflows while opening up the possibility of using linters that are not capable of processing Shellcheck's JSON output.

**Comment by koalaman** [comment]
A special format for one tool that supports a slight extensions to the GCC format is a bit too niche for a separate format. Perhaps it could be implemented by piping through `jq` instead?

**Comment by lcrockett** [comment]
Thank you for your suggestion. However, piping it through an external helper like `jq` will not resolve the fact that GCC formatting does not distinguish between `info` and `style` levels. This issue is about the lack GCC based formatting that supports all 4 levels instead of the historically used 3 levels.

**Comment by koalaman** [comment]
ShellCheck has no influence over GCC's output format, but if they change to 4 levels, then ShellCheck will definitely follow.
